sros2/api/__init__.py

```diff
@@ -306,32 +321,50 @@ def create_permission_file(path, name, domain_id, permissions_dict):
                 print("unknown permission policy '%s', skipping" % policy['allow'])
                 continue
             permission_str += """\
-      <%s>
-        <topic>%s</topic>
-      </%s>
-""" % (tag, topic_name, tag)
+        <%s>
+          <partitions>
+            <partition>%s</partition>
+          </partitions>
+          <topics>
+            <topic>%s</topic>
+          </topics>
+        </%s>
+""" % (tag, 'rt', topic_name, tag)
         # DCPS* is necessary for builtin data readers
         permission_str += """\
-      <subscribe>
-        <topic>DCPS*</topic>
-      </subscribe>
+        <subscribe>
+          <topics>
+            <topic>DCPS*</topic>
+          </topics>
+        </subscribe>
 """
     else:
         # no policy found: allow everything!
         permission_str += """\
-      <publish>
-        <topic>*</topic>
-      </publish>
-      <subscribe>
-        <topic>*</topic>
-      </subscribe>
+        <publish>
+          <partitions>
+            <partition>*</partition>
+          </partitions>
+          <topics>
+            <topic>*</topic>
+          </topics>
+        </publish>
+        <subscribe>
+          <partitions>
+            <partition>*</partition>
+          </partitions>
+          <topics>
+            <topic>*</topic>
+          </topics>
+        </subscribe>
 """
 
     permission_str += """\
-    </allow_rule>
-    <default>DENY</default>
-  </grant>
-</permissions>
+      </allow_rule>
+      <default>DENY</default>
```

Comment:
Never mind what I said @mikaelarguedas , from testing this branch, your default deny rule works just fine.

Reply:
cool :+1: